tests/test_pypi.py

```diff
@@ -151,6 +151,174 @@ def test_get_extra_requirements():
+    recipe["name"] == "dask"
+    assert "outputs" not in recipe
+
+    found = sorted(recipe["requirements"]["run"])
+    expected = list(base_requirements)
+    for lst in extras.values():
+        expected += lst
+    assert found == sorted(set(expected))
+
+    found = sorted(recipe["test"]["requires"])
+    expected = ["pip"]
+    assert found == expected
+
+    # all extras are included in the requirements except for the
+    # test requirements which are in the test section
+    config = Configuration(
+        name="dask",
+        version="2022.6.1",
+        extras_require_all=True,
+        extras_require_test="test",
+    )
+    recipe = GrayskullFactory.create_recipe("pypi", config)
+    recipe["name"] == "dask"
+    assert "outputs" not in recipe
+
+    found = sorted(recipe["requirements"]["run"])
+    expected = list(base_requirements)
+    for name, lst in extras.items():
+        if name != "test":
+            expected.append(expected.append(f"# Extra: {name}"))
+            expected += lst
+    assert found == sorted(set(expected))
+
+    found = sorted(recipe["test"]["requires"])
+    expected = sorted(["pip"] + extras["test"])
+    assert found == expected
+
+    # only "array" is included in the requirements
+    config = Configuration(
+        name="dask", version="2022.6.1", extras_require_include=("array",)
+    )
+    recipe = GrayskullFactory.create_recipe("pypi", config)
+    recipe["name"] == "dask"
+    assert "outputs" not in recipe
+
+    found = sorted(recipe["requirements"]["run"])
+    expected = list(base_requirements)
+    expected.append("# Extra: array")
+    expected += extras["array"]
+    assert found == sorted(set(expected))
+
+    found = sorted(recipe["test"]["requires"])
+    assert found == ["pip"]
+
+    # only "test" is included but in the test section
+    config = Configuration(
+        name="dask",
+        version="2022.6.1",
+        extras_require_all=True,
+        extras_require_exclude=set(extras) - {"test"},
+        extras_require_test="test",
+    )
+    recipe = GrayskullFactory.create_recipe("pypi", config)
+    recipe["name"] == "dask"
+    assert "outputs" not in recipe
+
+    found = sorted(recipe["requirements"]["run"])
+    expected = sorted(base_requirements)
+    assert found == expected
+
+    found = sorted(recipe["test"]["requires"])
+    expected = sorted(["pip"] + extras["test"])
+    assert found == expected
+
+    # only "test" is included in the test section
+    config = Configuration(
+        name="dask",
+        version="2022.6.1",
+        extras_require_all=True,
+        extras_require_exclude=set(extras) - {"test"},
+        extras_require_test="test",
+        extras_require_split=True,
+    )
+    recipe = GrayskullFactory.create_recipe("pypi", config)
+    recipe["name"] == "dask"
+    assert "outputs" not in recipe
+
+    found = sorted(recipe["requirements"]["run"])
+    expected = sorted(base_requirements)
+    assert found == expected
```

Comment:
```suggestion
    assert set(recipe["requirements"]["run"]) == set(base_requirements)
```

Reply:
it would be better to convert the requirements run into a list or set and compare with that, or compare with it directly
like
```python
assert recipe["requirements"]["run"] == ["python >=3.6", "# comment", "pytest"]
```
That should work. The problem with sorting the requirements run it is because we might have comments, and it is difficult to sort it when we have those things, so it is better to convert it to a proper list that the comments will be just regular strings and it will not be related to the yaml object.

Reply:
Done.